src/include/api/ivf_pq_index.h

```diff
@@ -114,6 +115,12 @@ class IndexIVFPQ {
           id_datatype_ = string_to_datatype(value);
         } else if (key == "partitioning_index_type") {
           partitioning_index_datatype_ = string_to_datatype(value);
+        } else if (key == "distance_metric") {
+          try {
+            distance_metric_ = parseAndValidateDistanceMetric(value);
+          } catch (const std::runtime_error& e) {
+            std::cerr << e.what() << std::endl;
+          }
```

Comment:
Why do we need `std::cerr << e.what() << std::endl;`? Can't we just let `parseAndValidateDistanceMetric(value)` throw?